pkg/controller/control.go

```diff
@@ -901,3 +917,117 @@ func (c *Controller) BackupReplicaMappingDelete(id string) error {
 	delete(c.backupList, id)
 	return nil
 }
+
+func freezeVolume(ne *iutil.NamespaceExecutor, volume string) (bool, error) {
+	devicePath := "/dev/longhorn/" + volume
+	mountPoint, err := findMountPoint(ne, devicePath)
+	if err != nil {
+		logrus.WithField("volume", volume).Warn("Failed to execute find mount point, cannot freeze potential filesystem")
+		return false, err
+	} else if mountPoint == "" {
+		logrus.WithField("volume", volume).Info("Volume is not mounted, no need to freeze filesystem")
+		return false, nil
+	}
+
+	if err = freezeFilesystem(ne, mountPoint); err != nil {
+		logrus.WithField("volume", volume).WithError(err).Errorf("Failed to freeze the filesystem at %v", mountPoint)
+		return false, err
+	}
+
+	logrus.WithField("volume", volume).Infof("Froze filesystem of volume mounted at %v", mountPoint)
+	return true, nil
+}
+
+func unfreezeVolume(ne *iutil.NamespaceExecutor, volume string) error {
+	devicePath := "/dev/longhorn/" + volume
+	mountPoint, err := findMountPoint(ne, devicePath)
+	if err != nil {
+		logrus.WithField("volume", volume).Warn("Failed to execute find mount point, cannot unfreeze potential filesystem")
+		return err
+	} else if mountPoint == "" {
+		logrus.WithField("volume", volume).Info("Volume is not mounted, no need to unfreeze filesystem")
+		return nil
+	}
+
+	if err = unfreezeFilesystem(ne, mountPoint); err != nil {
+		logrus.WithField("volume", volume).WithError(err).Errorf("Failed to unfreeze the filesystem at %v", mountPoint)
+		return err
+	}
+
+	logrus.WithField("volume", volume).Infof("Unfroze filesystem of volume mounted at %v", mountPoint)
+	return nil
+}
+
+func findMountPoint(ne *iutil.NamespaceExecutor, devicePath string) (string, error) {
```

Comment:
Not sure if we need to consider the case that there are multiple mount points on a device.

Reply:
I only look for the `--first-only` since we are freezing the filesystem not the mount point.
Additional mount points are irrelevant.